File: app/src/main/java/com/termux/data/Properties.java
package com.termux.data;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.HashMap;
import java.util.Map;
import java.util.function.BiConsumer;
import java.util.regex.Pattern;

public final class Properties {
    private static final Pattern PATTERN = Pattern.compile(" : ");
    private final Map<String, String> map = new HashMap<>(5);

    public Properties(final String filepath) {
        try {
            final BufferedReader br = new BufferedReader(new FileReader(filepath));
            String line;
            while (null != (line = br.readLine())) {
                final String[] args = Properties.PATTERN.split(line);
                if (2 == args.length) this.map.put(args[0], args[1]);
            }
        } catch (final Throwable ignored) {
        }
    }

    public void forEach(final BiConsumer<? super String, ? super String> action) {
        this.map.forEach(action);
    }

    String get(final String key) {
        return this.map.get(key);
    }

    public int getInt(final String key, final int defaultValue) {
        final var s = this.map.get(key);
        return null == s ? defaultValue : Integer.parseInt(s);
    }

    public boolean getBoolean(final String key, final boolean defaultValue) {
        final var s = this.map.get(key);
        return null == s ? defaultValue : Boolean.parseBoolean(s);
    }
}

File: app/src/main/java/com/termux/data/Properties.kt
package com.termux.data

import java.io.File

class Properties(filePath: String) {
    private val map = mutableMapOf<String, String>()

    init {
        val file = File(filePath)
        if (file.exists()) {
            file.forEachLine { line ->
                line.split(" : ").let { strings ->
                    if (strings.size != 2) return@let
                    map[strings[0]] = strings[1]
                }
            }
        }
    }

    fun forEach(action: (key: String, value: String) -> Unit): Unit = map.forEach(action)

    fun get(key: String): String? = map[key]

    fun getInt(key: String, default: Int): Int = get(key)?.toInt() ?: default

    fun getBoolean(key: String, default: Boolean): Boolean = get(key)?.toBoolean() ?: default
}
